internal/rancher/provisioning/v1/cluster.go

```diff
@@ -0,0 +1,34 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Cluster is the struct representing a Rancher Cluster.
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+type Cluster struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Status ClusterStatus `json:"status,omitempty"`
+}
+
+// ClusterStatus is the struct representing the status of a Rancher Cluster.
+type ClusterStatus struct {
+	ClusterName   string `json:"clusterName,omitempty"`
+	AgentDeployed bool   `json:"agentDeployed,omitempty"`
+	Ready         bool   `json:"ready,omitempty"`
+}
+
+// ClusterList contains a list of ClusterList.
+// +kubebuilder:object:root=true
+type ClusterList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []Cluster `json:"items"`
+}
+
+func init() {
+	SchemeBuilder.Register(&Cluster{}, &ClusterList{})
```

Comment:
suggestion: there has been a recent effort to avoid reliance on controller runtime schemeBuilder in CAPI (xref: https://github.com/kubernetes-sigs/cluster-api/pull/9045/files) which we could also follow

Reply:
I don't think it's required for our case but it's a good habit to start doing it like it's described in CAPI

Reply:
I don't think it's required for our case but it's a good habit to start doing it like it's described in CAPI

Reply:
I don't think it's required for our case but it's a good habit to start doing it like it's described in CAPI

Reply:
I think it is better if we align our code when we are at it and early, rather then leaving the work for future

Reply:
Who may "suggest" it and why? I would prefer having an issue describing why we'd likely want to avoid using controller runtime without an API, so this, if makes sense could be tracked separately. This PR has a concrete change in mind.

Reply:
it's done to be able to import API without relying on controller-runtime, in this case, we have these wrappers in an internal package so no other projects can import it. I totally agree with @furkatgofurov7 but I don't want to block this PR because of that.

Reply:
Fine as is for now by me